src/sdkBase.ts

```diff
@@ -174,33 +188,109 @@ export abstract class SdkBase {
   }
 
   private initInternal(opts: SdkOptions | unknown) {
-    if (this._initComplete) {
-      throw new TypeError('Calling init() more than once is not allowed');
-    }
     if (!isSDKOptionsOrThrow(opts))
       throw new TypeError(`Options provided to ${this._product.name} init couldn't be validated.`);
 
-    storeConfig(opts, this._product);
+    if (this._hasAborted) {
+      throw new TypeError('Calling init() once aborted or disconnected is not allowed');
+    }
 
-    this._opts = opts;
-    this._storageManager = new StorageManager(
-      { ...opts },
-      this._product.cookieName,
-      this._product.localStorageKey
-    );
-    this._apiClient = new ApiClient(opts, this._product.defaultBaseUrl, this._product.name);
-    this._tokenPromiseHandler.registerApiClient(this._apiClient);
+    if (this.isInitialized()) {
```

Comment:
This is a fairly long _if_ body, and I think you could offload a lot of the decisions to the various services - e.g. just give the API client the new config and let it work out if it needs to do anything. The services mostly store their config, so they'll already have the previous config if it's relevant.

There might be a few special cases - e.g. after you work out whether to use a new identity, you could pass the new identity to the storage manager update function.